File: 2006.差的绝对值为-k-的数对数目1.go
/*
 * @lc app=leetcode.cn id=2006 lang=golang
 *
 * [2006] 差的绝对值为 K 的数对数目
 */

// @lc code=start
func countKDifference(nums []int, k int) (res int) {
	// 哈希表加速访问
	nunmMap := map[int]int{}
	for _, num := range nums {
		res += numMap[nu-k] + numMap[num+k]
		numMap[num]++
	}
	return
}

// @lc code=end

File: 2006.差的绝对值为-k-的数对数目1.py
#
# @lc app=leetcode.cn id=2006 lang=python3
#
# [2006] 差的绝对值为 K 的数对数目
#

# @lc code=start
class Solution:
    def countKDifference(self, nums: List[int], k: int) -> int:
        # 哈希表记录加速访问
        res = 0
        num_dict = {}
        for num in nums:
            if num - k in num_dict:
                res += num_dict[num - k]
            if num + k in num_dict:
                res += num_dict[num + k]
            
            if num in num_dict:
                num_dict[num] += 1
            else:
                num_dict[num] = 1
        return res
# @lc code=end

